### PR TITLE
remotexpc: refactor to use asyncio

### LINKED ISSUE #739: tunneld hangs when pair dialog is shown
**Test environment**
- Host OS version: Ventura 13.6
- Target device model and iOS version: all iOS 17.x devices

**Describe the bug**
Upon launching tunneld daemon, when several devices are connected and at least 1 is not paired to the mac, all other paired devices cannot take a screenshot.

**To Reproduce**
1. Connect several devices thru a usb hub and unpair 1 device
2. Execute: `sudo pymobiledevice3 remote tunneld -d`
3. Take a screenshot of a paired device: `pymobiledevice3 developer dvt screenshot ./test.png --tunnel <serial>`
4. The screenshot command will not complete and just hang up

**Expected behavior**
Tunneld should disregard unpaired devices and not affect all others when executing commands.

### PR DESCRIPTION
**Pull request by doronz88**
Close #739

### CODE REVIEW

File: pymobiledevice3/remote/remotexpc.py [hunk 1]
```diff
@@ -35,32 +34,34 @@ class RemoteXPCConnection:
...
+        await self.service_connection.aio_sendall(DataFrame(stream_id=ROOT_CHANNEL, data=xpc_wrapper).serialize())
 
-    def iter_file_chunks(self, total_size: int, file_idx: int = 0) -> Generator[bytes, None, None]:
+    async def iter_file_chunks(self, total_size: int, file_idx: int = 0) -> Generator[bytes, None, None]:
         stream_id = (file_idx + 1) * 2
-        self._open_channel(stream_id, XpcFlags.FILE_TX_STREAM_RESPONSE)
+        await self._open_channel(stream_id, XpcFlags.FILE_TX_STREAM_RESPONSE)
         size = 0
         while size < total_size:
             frame = self._receive_next_data_frame()
```
Comment (netanelc305) [inline pymobiledevice3/remote/remotexpc.py]: missing `await`

File: pymobiledevice3/remote/tunnel_service.py [hunk 2]
```diff
@@ -469,35 +474,36 @@ def pair_record_path(self) -> Path:
...
-        response = self._receive_plain_response()['event']['_0']
+        response = await self._receive_plain_response()
+        response = response['event']['_0']
 
         if 'pairingRejectedWithError' in response:
             raise PairingError(
                 response['pairingRejectedWithError']['wrappedError']['userInfo']['NSLocalizedDescription'])
         elif 'awaitingUserConsent' in response:
-            pairingData = self._receive_pairing_data()
+            pairingData = await self._receive_pairing_data()
```
Comment (netanelc305) [inline pymobiledevice3/remote/tunnel_service.py]: should be `pairing_data` ?

File: pymobiledevice3/remote/tunnel_service.py [hunk 3]
```diff
@@ -779,21 +786,26 @@ def __init__(self, rsd: RemoteServiceDiscoveryService):
...
-    def send_request(self, data: Mapping) -> None:
-        return self.service.send_request({
+    async def connect(self, autopair: bool = True) -> None:
+        await RemoteService.connect(self)
+        try:
+            response = await self.service.receive_response()
+            self.version = response['ServiceVersion']
+            await RemotePairingProtocol.connect(self, autopair=autopair)
+            self.hostname = self.service.address[0]
+        except:  # noqa: E722
```
Comment (netanelc305) [inline pymobiledevice3/remote/tunnel_service.py:796]: No specific exception?

File: pymobiledevice3/service_connection.py [hunk 4]
```diff
@@ -169,10 +173,13 @@ async def aio_recv_plist(self, endianity='>') -> bytes:
     def send_plist(self, d, endianity='>', fmt=plistlib.FMT_XML) -> None:
         return self.sendall(build_plist(d, endianity, fmt))
 
-    async def aio_send_plist(self, d, endianity='>', fmt=plistlib.FMT_XML) -> None:
-        self.writer.write(build_plist(d, endianity, fmt))
+    async def aio_sendall(self, payload: bytes) -> None:
+        self.writer.write(payload)
         await self.writer.drain()
 
+    async def aio_send_plist(self, d, endianity='>', fmt=plistlib.FMT_XML) -> None:
```
Comment (netanelc305) [inline pymobiledevice3/service_connection.py]: Add type hints

**Review by netanelc305 — CHANGES_REQUESTED** [review]

File: pymobiledevice3/remote/remotexpc.py (hunk 1 above)
Comment (doronz88) [reply pymobiledevice3/remote/remotexpc.py]: done

File: pymobiledevice3/remote/tunnel_service.py (hunk 2 above)
Comment (doronz88) [reply pymobiledevice3/remote/tunnel_service.py]: done

File: pymobiledevice3/remote/tunnel_service.py (hunk 3 above)
Comment (doronz88) [reply pymobiledevice3/remote/tunnel_service.py:796]: discussed

File: pymobiledevice3/service_connection.py (hunk 4 above)
Comment (doronz88) [reply pymobiledevice3/service_connection.py]: done

**Pull request merged**